src/mail/view/MailView.js

```diff
@@ -513,16 +513,16 @@ export class MailView implements CurrentView {
 	 *
 	 * @param args Object containing the optional parts of the url which are listId and mailId for the mail view.
 	 */
-	updateUrl(args: Object) {
-		if (m.route.get().indexOf("/mailto") === 0) {
-			if (location.hash.length > 5) {
-				let url = location.hash.substring(5)
-				let decodedUrl = decodeURIComponent(url)
+	updateUrl(args: Object, requestedPath: string) {
+		if (requestedPath.startsWith("/mailto")) {
+			const query = m.parseQueryString(location.hash.substring(1)) // eat the leading hash
+			if (query.url) {
+				const decodedUrl = decodeURIComponent(query.url)
 				Promise.all([locator.mailModel.getUserMailboxDetails(), import("../editor/MailEditor")])
 				       .then(([mailboxDetails, {newMailtoUrlMailEditor}]) => {
 					       newMailtoUrlMailEditor(decodedUrl, false, mailboxDetails).then(editor => editor.show())
-					       history.pushState("", document.title, window.location.pathname) // remove # from url
```

Comment:
Will the hash stay now?

Reply:
it does stay. I've just played around with it again and it's behaving differently from how it was yesterday for some reason as well. Not sure now that this change would be consistently able to fix the issue...

Reply:
It's just so that we don't keep the data in the URL and that if you reload it doesn't appear again but idk if we can replace original history entry.

Reply:
There might be another way around that. It stops being an issue if the url changes, like from closing the editor.

For some reason now when i'm trying it out though, clicking back is causing the editor to pop directly back up, which wasn't happening yesterday >.> it might be from having done a clean build or something. so this needs some more work anyway.